### PR TITLE
[WIP] Forms FE

### PR DESCRIPTION
**Pull request by nicklee**
https://projects.torchbox.com/projects/rca-website-rebuild/tickets/652

### CODE REVIEW

**Comment by nicklee** [comment]
Rebased this on master, and it seems to have included some of Tibors items, feel free to ignore them, not entirely sure how they've ended up here.

File: rca/project_styleguide/templates/patterns/atoms/form-checkbox/form-checkbox.html [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+                    <input class="form-item__checkbox" type="checkbox" id="{{ field.id_for_label }}" {% if field.field.required %}required{% endif %}>
+                    {{ field.label }}
+                    {% if field.field.required %} <span class="form-item__required">*</span>{% endif %}
+                </label>
+            </li>
+        </ul>
+
+        
+
+    </div>
```
Comment (chris-lawton) [inline rca/project_styleguide/templates/patterns/atoms/form-checkbox/form-checkbox.html:16]: nitpick: could tidy up the space here

File: rca/project_styleguide/templates/patterns/pages/forms/form_page.html [hunk 2]
```diff
@@ -1,21 +1,45 @@
 {% extends "patterns/base_page.html" %}
 {% load wagtailcore_tags wagtailimages_tags util_tags static %}
 
+{% block body_class %}
+    app--form
+{% endblock %}
+
+{% block html_class %}sticky-compensator{% endblock %}
```
Comment (chris-lawton) [inline rca/project_styleguide/templates/patterns/pages/forms/form_page.html:8]: 👌

File: rca/static_src/javascript/components/form-focus.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    static selector() {
+        return '[data-focus]';
+    }
+
+    constructor(node) {
+        this.formItem = node;
+        this.formFieldInput = this.formItem.querySelector('input');
+        this.formFieldTextarea = this.formItem.querySelector('textarea');
+        this.formType = this.formItem.dataset.focustype;
+        this.focusClass = 'form-item--has-focus';
```
Comment (chris-lawton) [inline rca/static_src/javascript/components/form-focus.js:11]: could name this `hasFocusClass` to be consistent with `hasContentClass`

**Review by chris-lawton — APPROVED** [review]

File: rca/project_styleguide/templates/patterns/atoms/form-checkbox/form-checkbox.html (hunk 1 above)
Comment (nicklee) [reply rca/project_styleguide/templates/patterns/atoms/form-checkbox/form-checkbox.html:13]: 👍

**Pull request merged**